### ISSUE TITLE
Large hex_string of pickle.load() crashes gpython

### ISSUE THREAD

**Issue by xiaxinmeng** [go-python/gpython#201]
When pickle loads large string of a byte file, gpython crashes with segfault

Crashing message: segmentation fault
 
```
import io
import pickle
hex_string = '8004952A000000000000008C086461746574696D65948C086461746574696D65949388430A07B2010100000000000092059452942E'
myb = bytes.fromhex(hex_string)
f = io.BytesIO(myb)
myb = pickle.load(f)
```

environment: gpython 0.1.0 on Ubuntu 18.04, and gpython main(https://github.com/go-python/gpython/commit/6f8e06a4660709ab44398d8b1a18738aa407b1c3 on Oct 5)

**Comment by xiaxinmeng** [comment]
I am very sorry. After checking again, I think I made mistakes in building gpython from source.  I downloaded source code from GitHub and release websites(https://github.com/go-python/gpython/releases). Then i built gpython with the following command: cd go-python/gpython/bin/; ./install_python.sh xxxdir. And hence, I had mistaken Python 3.4.9 for gpython. Yes, indeed, these test programs cannot crash gpython, just crash CPython 3.4.9. 

>  where did you get these from?

Our team is developing  a fuzzing tool to detect bugs in python interpreters. We hope that these bugs can be helpful for the development of gpython. Sorry again, we will carefully check the bugs in later reports. Thanks.
